flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/operators/StreamSourceOperatorLatencyMetricsTest.java

```diff
@@ -164,7 +166,10 @@ private void testLatencyMarkEmission(int numberLatencyMarkers, OperatorSetupOper
 		operatorSetup.setupSourceOperator(operator, testProcessingTimeService);
 
 		// run and wait to be stopped
-		operator.run(new Object(), mock(StreamStatusMaintainer.class), new CollectorOutput<Long>(output), mock(OperatorChain.class));
+		OperatorChain<?, ?> operatorChain = new OperatorChain<>(
+			operator.getContainingTask(),
+			StreamTask.createRecordWriters(operator.getOperatorConfig(), new MockEnvironmentBuilder().build()));
+		operator.run(new Object(), mock(StreamStatusMaintainer.class), new CollectorOutput<Long>(output), operatorChain);
```

Comment:
I think you should add `try/finally` block with `operatorChain.releaseOutputs()`, otherwise there will be a resource leak.

Reply:
You're right. I overlooked that.

Reply:
Thanks for fixing this also in other places.